test/parallel/test-net-better-error-messages-port.js

```diff
@@ -3,12 +3,14 @@ const common = require('../common');
 const net = require('net');
 const assert = require('assert');
 
-const c = net.createConnection(common.PORT);
-
+const server = net.createServer();
+server.listen(0);
+const port = server.address().port;
+const c = net.createConnection(port);
 c.on('connect', common.mustNotCall());
-
-c.on('error', common.mustCall(function(e) {
+c.on('error', common.mustCall((e) => {
   assert.strictEqual(e.code, 'ECONNREFUSED');
-  assert.strictEqual(e.port, common.PORT);
+  assert.strictEqual(e.port, port);
   assert.strictEqual(e.address, '127.0.0.1');
 }));
+server.close();
```

Comment:
> @santigimeno close() is actually executed synchronously, but by the time the connection attempt is made (which happens asynchronously), the port is still up for grab, right? Perhaps I am overthinking 🤔

Yes, I thought of that too. And I think I'm also overthinking 😆 .

> I am afraid sequential also will not solve this. common.PORT could still have been assigned to other processes.

Right, but not to other tests in the same run though.

> Perhaps we can live with the current change...

Yeah, probably they'll never fail.